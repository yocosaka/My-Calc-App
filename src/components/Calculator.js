import React from 'react';
import App from './App';

const Calculator = () => (
  <div className="main-content">
    <h2 className="main-title">Let&apos;s do some math!</h2>
    <App />
  </div>
);

export default Calculator;
