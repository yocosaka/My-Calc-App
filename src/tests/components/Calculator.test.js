import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Calculator from '../../components/Calculator';

describe('Calculator', () => {
  test('renders Calculator component ', () => {
    render(<Calculator />);
    // eslint-disable-next-line no-unused-expressions
    expect(screen.getByText("Let's do some math!")).toBeInTheDocument;
  });
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Calculator />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
