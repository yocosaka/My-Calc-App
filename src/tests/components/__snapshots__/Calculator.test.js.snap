// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should match with snapshot 1`] = `
<div
  className="main-content"
>
  <h2
    className="main-title"
  >
    Let's do some math!
  </h2>
  <div
    className="calculator"
  >
    <div
      className="display"
    >
      <span>
        0
      </span>
    </div>
    <div
      className="btns"
    >
      <div
        className="btn__group"
      >
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="btn__group"
      >
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="btn__group"
      >
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="btn__group"
      >
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="btn__group"
      >
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
